src/wallet/wallet.cpp

```diff
@@ -767,17 +767,6 @@ bool CWallet::EncryptWallet(const SecureString& strWalletPassphrase)
         delete pwalletdbEncryption;
         pwalletdbEncryption = nullptr;
 
-        Lock();
-        Unlock(strWalletPassphrase);
-
-        // if we are using HD, replace the HD master key (seed) with a new one
-        if (IsHDEnabled()) {
-            if (!SetHDMasterKey(GenerateNewHDMasterKey())) {
-                return false;
-            }
-        }
-
-        NewKeyPool();
```

Comment:
Does NewKeyPool delete existing key and that's why you removed it?

Reply:
Yep, it removes (marks as used) old keys from the pool and topups it again